http-server/src/server.rs

```diff
@@ -344,6 +344,23 @@ impl<M> Builder<M> {
 			health_api: self.health_api,
 		})
 	}
+
+	/// Pre-build the server as is.
+	pub fn pre_build(self) -> Server<M> {
+		Server {
+			listener: None,
+			local_addr: None,
+			access_control: self.access_control,
+			max_request_body_size: self.max_request_body_size,
+			max_response_body_size: self.max_response_body_size,
+			batch_requests_supported: self.batch_requests_supported,
+			resources: self.resources,
+			tokio_runtime: self.tokio_runtime,
+			middleware: self.middleware,
+			max_log_length: self.max_log_length,
+			health_api: self.health_api,
+		}
+	}
```

Comment:
Could you just convert the builder straight into the `RPSeeServerSvc` without needing this "pre-build" step (which I assume doesn't really make any sense outside of creating a service?)

(also we'd want to tweak the names ultimately; `RPSeeServerSvc` is quite hideous :D but that's not important here!)